### PR TITLE
[Feature] Add EAST head

### PR DESCRIPTION
**Pull request by Mountchicken**
- Modules
- [x] EAST neck
- [x] EAST head
- [ ] EAST loss
- [ ] EAST postprocessor
- [ ] EAST Targets
- Unit Tests
- [x] EAST neck
- [ ] EAST head
- [ ] EAST loss
- [ ] EAST postprocessor
- [ ] EAST Targets

### CODE REVIEW

File: mmocr/models/textdet/dense_heads/east_head.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self.num_classes = num_classes
+        self.in_channels = in_channels
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self._init_layers()
+
+    def _init_layers(self):
+        """Initialize layers of the head."""
+        self._init_cls_convs()
+        self._init_reg_convs()
```
Comment (xinke-wang) [inline mmocr/models/textdet/dense_heads/east_head.py:42]: These `_init_layers`, `_init_cls_convs`, and `_init_reg_convs` look a little bit weird here. Is it necessary to split these definitions into two functions? When I look at the names, my first impression is that these functions are used for initializing weights. Directly defining `self.score_conv` and `self.rec_conv` in `__init__` might be enough

File: mmocr/models/textdet/dense_heads/east_head.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    def _init_cls_convs(self):
+        """Initialize score prediction conv layers of the head."""
+        self.score_conv = Sequential(
+            ConvModule(
+                self.in_channels,
+                out_channels=1,
+                kernel_size=1,
+                stride=1,
+                norm_cfg=dict(type='BN'),
+                act_cfg=dict(type='ReLU')))
```
Comment (xinke-wang) [inline mmocr/models/textdet/dense_heads/east_head.py:74]: The [official implementation](https://github.com/argman/EAST/blob/master/model.py#L74) uses `Sigmoid` activation in all output layers.
```suggestion
                act_cfg=dict(type='Sigmoid')))
```

File: mmocr/models/textdet/dense_heads/east_head.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        """Initialize bbox regression conv layers of the head."""
+        if self.box_type == 'RBOX':
+            self.ang_conv = Sequential(
+                ConvModule(
+                    self.in_channels,
+                    out_channels=1,
+                    kernel_size=1,
+                    stride=1,
+                    norm_cfg=dict(type='BN'),
+                    act_cfg=dict(type='ReLU')))
```
Comment (xinke-wang) [inline mmocr/models/textdet/dense_heads/east_head.py:54]: The [official implementation](https://github.com/argman/EAST/blob/master/model.py#L74) uses `Sigmoid` activation in all output layers.
```suggestion
                    act_cfg=dict(type='Sigmoid')))
```

File: mmocr/models/textdet/dense_heads/east_head.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        out_channels = 4 if self.box_type == 'RBOX' else 8
+        self.reg_conv = ConvModule(
+            self.in_channels,
+            out_channels,
+            kernel_size=1,
+            stride=1,
+            norm_cfg=dict(type='BN'),
+            act_cfg=dict(type='ReLU'))
```
Comment (xinke-wang) [inline mmocr/models/textdet/dense_heads/east_head.py:63]: The [official implementation](https://github.com/argman/EAST/blob/master/model.py#L74) uses `Sigmoid` activation in all output layers.
```suggestion
            act_cfg=dict(type='Sigmoid'))
```

File: mmocr/models/textdet/dense_heads/east_head.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+from mmcv.runner import BaseModule, Sequential
+
+from mmocr.models.builder import HEADS
+from .head_mixin import HeadMixin
+
+
+@HEADS.register_module()
+class EASTHead(HeadMixin, BaseModule):
+
+    def __init__(self,
```
Comment (xinke-wang) [inline mmocr/models/textdet/dense_heads/east_head.py:12]: No docstring at current version, may need to be added later.

**Comment by xinke-wang** [comment]
Unittest needs to be added later.

File: mmocr/models/textdet/dense_heads/east_head.py (hunk 1 above)
Comment (Mountchicken) [reply mmocr/models/textdet/dense_heads/east_head.py:42]: You are right, This is indeed weird.

**Comment by Mountchicken** [comment]
Still working on the loss non-drop problem. When it can be reproduced, I'll solve problems above.

**Pull request closed without merging**